src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt

```diff
@@ -328,6 +328,34 @@ class BatchStore(
     }
   }
 
+  fun changeStatuses(
+      batchId: BatchId,
+      germinatingQuantityToChange: Int,
+      notReadyQuantityToChange: Int
+  ) {
+    requirePermissions { updateBatch(batchId) }
+
+    if (germinatingQuantityToChange == 0 && notReadyQuantityToChange == 0) {
+      return
+    }
+
+    retryVersionedBatchUpdate(batchId) { batch ->
+      if (batch.germinatingQuantity < germinatingQuantityToChange ||
+          batch.notReadyQuantity < notReadyQuantityToChange) {
+        throw BatchInventoryInsufficientException(batchId)
+      }
```

Comment:
is this within the retry so we can capture issues from concurrent edits?

Reply:
Yes,  exactly -- someone else might have withdrawn or changed statuses such that the current status change can no longer succeed.